include/hwinfo.h

```diff
@@ -0,0 +1,54 @@
+/**
+ * @file
+ *
+ * @brief Public APIs to get device Information.
+ */
+
+/*
+ * Copyright (c) 2018 Alexander Wachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_HWINFO_H_
+#define ZEPHYR_INCLUDE_HWINFO_H_
+
+/**
```

Comment:
The includes of system files must be outside the `extern "C"` block.  I think the right order is includes, `extern "C"`, then doxygen open, declarations, doxygen close, `extern` close, then next include.

See 68b1890ead8cfbbc371bdd31be4bfc80abc814f5

Reply:
Done